aws/policy/data-services.yaml

```
@@ -75,6 +75,7 @@ Statement:
       - rds:RestoreDBInstanceFromS3
       - rds:CreateDBInstanceReadReplica
       - rds:CreateDBInstance
+      - rds:CreateTenantDatabase
```

```suggestion
      - rds:*TenantDatabase
```
to support `ModifyTenantDatabase` and `DeleteTenantDatabase`

---

We should avoid wildcarding in this way. If there's a need to reduce the size of this policy there are better ways to do it.

---

I was about to say that. We add wildcarding for List or Get permissions which are harmless. I'm testing right now with the terminator and I would only stick to the permissions needed for that PR.

---

@gravesm  It seems to me what we do not need a dedicated class. I tried to force the cleanup.

```
(py311) alinabuzachis@Alinas-MacBook-Pro aws % python3 cleanup.py  --stage dev --target RdsDbInstance --force

cleanup     : INFO     terminated RdsDbInstance: name=ansible-test-oracle-339547aee27a, id=arn:aws:rds:us-east-1:xxx:db:ansible-test-oracle-339547aee27a age=0:03:31, stale=False
cleanup     : INFO     terminated RdsDbInstance: name=ansible-test-oracle-8581d1fe8fa7, id=arn:aws:rds:us-east-1:xxx:db:ansible-test-oracle-8581d1fe8fa7 age=0:03:30, stale=False
```
And if I re-run the same command I get for both
`.InvalidDBInstanceStateFault: An error occurred (InvalidDBInstanceState) when calling the ModifyDBInstance operation: The specified database instance is currently in deleting state, which is a terminal state."}.InvalidDBInstanceStateFault: An error occurred (InvalidDBInstanceState) when calling the ModifyDBInstance operation: The specified database instance is currently in deleting state, which is a terminal state."}`

Then after some time:
```
(py311) alinabuzachis@Alinas-MacBook-Pro aws % python3 cleanup.py  --stage dev --target RdsDbInstance --force -vvvv

cleanup     : DEBUG    Config path: /Users/alinabuzachis/dev/aws-terminator/aws/config.yml
cleanup     : DEBUG    located RdsDbInstance: count=0
```

I also tried:
```
aws rds describe-tenant-databases --region us-east-1     
{
    "TenantDatabases": []
}
```

---

It seems @GomathiselviS has added some units tests for the functionality https://github.com/ansible-collections/amazon.aws/pull/2275 and removed the integration tests. @gravesm Do we still want to add merge this change or can we close this PR? I'll be inclined to close it.